package/gluon-migrate-vpn/files/lib/gluon/upgrade/499-migrate-vpn

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/lua
+
+local site = require 'gluon.site_config'
+local users = require 'gluon.users'
+local util = require 'gluon.util'
+
+local uci = require('luci.model.uci').cursor()
+local lutil = require 'luci.util'
+
+local tunneldigger_enabled = uci:get_first('tunneldigger', 'broker', 'enabled')
+local fastd_enabled = uci:get('fastd', 'mesh_vpn', 'enabled')
+
+local tunneldigger_installed = util.exec('sh' , '-c', 'opkg list-installed | grep -e \'^tunneldigger\'')
+local fastd_installed = util.exec('sh' , '-c', 'opkg list-installed | grep -e \'^fastd\'')
```

Comment:
This will break on systems without opkg (for example the *ar71xx-tiny* target of the Gluon *lede* branch). I think testing for the existence of `/usr/bin/fastd` and `/usr/bin/tunneldigger` would be more robust.